a_test.go

```diff
@@ -183,42 +186,360 @@ func TestA_uses_Logger_dynamic_interface(t *testing.T) {
+			assertFail(t, err, "flow execute")
+			assertErrorContains(t, err, "task failed", "must be a failed error")
+
+			for index, result := range childTasksResults {
+				chkFunc := assertTrue
+				if !c.parallel && index != 0 { // only first will be true if not parallel and with panic
+					chkFunc = assertFalse
+				}
+
+				chkFunc(t, result, "child result "+strconv.Itoa(index)+" is incorrect")
+			}
+
+			expectedFatalCalls := 1
+			if c.parallel {
+				expectedFatalCalls = len(childTasks)
+			}
+
+			assertTrue(t, loggerSpy.called, "logger call")
+			assertEqual(t, loggerSpy.fatalCallCount, expectedFatalCalls, "fatal calls count")
+		})
+	}
+}
+
+func TestA_WithContextSkipped(t *testing.T) {
+	type ctxKeyT struct{}
+	var ctxKey ctxKeyT
+
+	cases := []struct {
+		name     string
+		parallel bool
+	}{
+		{
+			name:     "parallel",
+			parallel: true,
+		},
+		{
+			name: "sequential",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			flow, task, childTasksResults, loggerSpy := prepareFlowAndTask(t, 3, ctxKey, func(a *goyek.A, i int) {
+				a.Skip(i)
+
+				a.Cleanup(func() {
+					skipped := a.Skipped()
+					assertTrue(t, skipped, "a.Skipped() should return true for "+a.Name())
+				})
+			})
+			childTasks := prepareTasks(t, flow, len(childTasksResults), c.parallel, task, func(i int) func(a *goyek.A) {
+				return func(a *goyek.A) {
+					a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+					call := onceCall(t, a.Name())
+
+					call()
+					prevCtx := a.Context()
+					newA := a.WithContext(context.WithValue(prevCtx, ctxKey, i))
+					assertEqual(t, newA.Name(), a.Name(), "name changed for "+a.Name())
+					assertEqual(t, a.Context(), prevCtx, "context changed for "+a.Name())
+
+					task.Action()(newA)
+
+					call()
+					a.Cleanup(func() {
+						skipped := a.Skipped()
+						assertTrue(t, skipped, "a.Skipped() should return true for "+a.Name())
+					})
+				}
+			})
+
+			main := flow.Define(goyek.Task{
+				Name: "main",
+				Deps: childTasks,
+			})
+
+			err := flow.Execute(context.Background(), []string{main.Name()})
+			assertPass(t, err, "flow execute")
+
+			for index, result := range childTasksResults {
+				assertTrue(t, result, "child result "+strconv.Itoa(index)+" is incorrect")
+			}
+
+			assertTrue(t, loggerSpy.called, "logger call")
+			assertEqual(t, loggerSpy.skipCallCount, len(childTasks), "skip calls count")
+		})
+	}
+}
+
+func prepareFlowAndTask(t *testing.T, childCount int, ctxKey interface{}, additionalAction func(*goyek.A, int)) (*goyek.Flow, *goyek.DefinedTask, []bool, *helperLoggerSpy) {
+	t.Helper()
+
+	flow := &goyek.Flow{}
+
+	flow.SetOutput(io.Discard)
+	loggerSpy := &helperLoggerSpy{}
+	flow.SetLogger(loggerSpy)
+
+	depsTask := flow.Define(goyek.Task{
+		Name: "deps",
+		Action: func(a *goyek.A) {
+			a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+		},
+	})
+
+	childTasksResults := make([]bool, childCount)
+	return flow, flow.Define(goyek.Task{
```

Comment:
Is there any reason to return `*goyek.DefinedTask` instead of the action `func(*goyek.A)`?
Does it have any relevance to `A.WithContext`?